diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e28815c4..25ee857fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,7 +291,7 @@ if(RDK_BUILD_PYTHON_WRAPPERS)
 
   #-------
   # pull in python:
-  find_package(Python3 COMPONENTS Interpreter Development.Module NumPy)
+  find_package(Python3 COMPONENTS Interpreter Development NumPy)
   target_include_directories(rdkit_py_base INTERFACE ${Python3_INCLUDE_DIRS})
   target_include_directories(rdkit_py_base INTERFACE ${Python3_NumPy_INCLUDE_DIRS})
 