From aab96d790af3200df929c91da3c673a2f8a54fd0 Mon Sep 17 00:00:00 2001
From: Steven Kearnes <skearnes@gmail.com>
Date: Mon, 28 Oct 2024 11:06:44 -0400
Subject: [PATCH] Remove .Module

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 705e292b2..ddaaf5694 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,7 +303,7 @@ if(RDK_BUILD_PYTHON_WRAPPERS)
 
   #-------
   # pull in python:
-  find_package(Python3 COMPONENTS Interpreter Development.Module NumPy)
+  find_package(Python3 COMPONENTS Interpreter Development NumPy)
   target_include_directories(rdkit_base INTERFACE ${Python3_INCLUDE_DIRS})
   target_include_directories(rdkit_base INTERFACE ${Python3_NumPy_INCLUDE_DIRS})
 
-- 
2.39.5 (Apple Git-154)

