diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4066d7e3e..518ba24df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -616,6 +616,7 @@ install(
   FILE ${RDKit_EXPORTED_TARGETS}.cmake
   NAMESPACE RDKit::
   DESTINATION ${RDKit_LibDir}/cmake/rdkit
+  COMPONENT dev
 )
 
 # create and install package configuration and version files
@@ -632,7 +633,8 @@ configure_file (
 install(FILES
     ${RDKit_BINARY_DIR}/rdkit-config.cmake
     ${RDKit_BINARY_DIR}/rdkit-config-version.cmake
-    DESTINATION ${RDKit_LibDir}/cmake/rdkit)
+    DESTINATION ${RDKit_LibDir}/cmake/rdkit
+    COMPONENT dev)
 
 # then manage the targets for the python bindings
 if(RDK_BUILD_PYTHON_WRAPPERS)
@@ -641,6 +643,7 @@ if(RDK_BUILD_PYTHON_WRAPPERS)
     FILE ${RDKitPython_EXPORTED_TARGETS}.cmake
     NAMESPACE RDKit::
     DESTINATION ${RDKit_LibDir}/cmake/rdkitpython
+    COMPONENT dev
   )
 
   write_basic_package_version_file(
@@ -656,7 +659,8 @@ if(RDK_BUILD_PYTHON_WRAPPERS)
   install(FILES
     ${RDKit_BINARY_DIR}/rdkitpython-config.cmake
     ${RDKit_BINARY_DIR}/rdkitpython-config-version.cmake
-    DESTINATION ${RDKit_LibDir}/cmake/rdkitpython)
+    DESTINATION ${RDKit_LibDir}/cmake/rdkitpython
+    COMPONENT dev)
 endif(RDK_BUILD_PYTHON_WRAPPERS)
 
 # Memory testing setup
diff --git a/Code/RDGeneral/CMakeLists.txt b/Code/RDGeneral/CMakeLists.txt
index c05545310..4f302df01 100644
--- a/Code/RDGeneral/CMakeLists.txt
+++ b/Code/RDGeneral/CMakeLists.txt
@@ -55,7 +55,8 @@ rdkit_headers(Exceptions.h
         DEST RDGeneral)
 
 if (NOT RDK_INSTALL_INTREE)
-    install(DIRECTORY hash DESTINATION ${RDKit_HdrDir}/RDGeneral)
+    install(DIRECTORY hash DESTINATION ${RDKit_HdrDir}/RDGeneral
+            COMPONENT dev)
 endif (NOT RDK_INSTALL_INTREE)
 
 rdkit_catch_test(testDict testDict.cpp LINK_LIBRARIES RDGeneral)
