diff --git a/Code/GraphMol/MolHash/hashfunctions.cpp b/Code/GraphMol/MolHash/hashfunctions.cpp
index 972d5b9..3a75506 100644
--- a/Code/GraphMol/MolHash/hashfunctions.cpp
+++ b/Code/GraphMol/MolHash/hashfunctions.cpp
@@ -680,7 +680,7 @@ static std::string ArthorSubOrderHash(RWMol *mol) {
   return buffer;
 }
 
-std::string MolHash(RWMol *mol, enum HashFunction func) {
+std::string MolHash(RWMol *mol, HashFunction func) {
   std::string result;
   char buffer[32];
   NMRDKitSanitizeHydrogens(mol);
diff --git a/Code/GraphMol/MolHash/nmmolhash.h b/Code/GraphMol/MolHash/nmmolhash.h
index 84479ac..3e173f6 100644
--- a/Code/GraphMol/MolHash/nmmolhash.h
+++ b/Code/GraphMol/MolHash/nmmolhash.h
@@ -42,7 +42,7 @@ enum class RDKIT_MOLHASH_EXPORT HashFunction {
   ArthorSubstructureOrder = 17
 };
 
-RDKIT_MOLHASH_EXPORT std::string MolHash(RWMol *mol, enum HashFunction func);
+RDKIT_MOLHASH_EXPORT std::string MolHash(RWMol *mol, HashFunction func);
 
 enum class RDKIT_MOLHASH_EXPORT StripType {
   AtomStereo = 1,
