diff --git a/CMakeLists.txt b/CMakeLists.txt
index 486c73619..5a38600e3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,9 +58,16 @@ option(RDK_INSTALL_DEV_COMPONENT "install libraries and headers" ON)
 option(RDK_USE_BOOST_REGEX "use boost::regex instead of std::regex (needed for systems with g++-4.8)" OFF)
 option(RDK_USE_BOOST_IOSTREAMS "use boost::iostreams" ON)
 option(RDK_BUILD_MINIMAL_LIB "build the minimal RDKit wrapper (for the JS bindings)" OFF)
+
 if(NOT MSVC)
   if(RDK_OPTIMIZE_POPCNT)
+    if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "aarch64")
+      message("RDK_OPTIMIZE_POPCNT is not available on aarch64")
+    elseif(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "ppc64le")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mpopcntb")
+    else()
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mpopcnt")
+    endif()
   endif()
 endif()
 
