From c003156bc3e3236fc46b60152a757aea7de49800 Mon Sep 17 00:00:00 2001
From: Steven Kearnes <skearnes@gmail.com>
Date: Wed, 24 Jul 2024 20:23:37 -0400
Subject: [PATCH] patch

---
 Code/PgSQL/rdkit/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Code/PgSQL/rdkit/CMakeLists.txt b/Code/PgSQL/rdkit/CMakeLists.txt
index 5a66ac4dd..8e81bd007 100644
--- a/Code/PgSQL/rdkit/CMakeLists.txt
+++ b/Code/PgSQL/rdkit/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(APPLE)
   set (EXTENSION_PREFIX "")
   set (EXTENSION_SUFFIX ".so")
-  set (EXTENSION_DEST_SUFFIX ".so")
+  set (EXTENSION_DEST_SUFFIX ".dylib")
 elseif(WIN32)
   set(REGEX_SEPARATOR "\\\\")
   if(MSVC)
-- 
2.39.3 (Apple Git-146)

