From 0b63838cc38c4256593aaa69d8767e9013032a3c Mon Sep 17 00:00:00 2001
From: Steven Kearnes <skearnes@gmail.com>
Date: Mon, 28 Oct 2024 08:28:01 -0400
Subject: [PATCH] Add find_package

---
 Code/cmake/Modules/RDKitUtils.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Code/cmake/Modules/RDKitUtils.cmake b/Code/cmake/Modules/RDKitUtils.cmake
index cf3a3b637..b1291943a 100644
--- a/Code/cmake/Modules/RDKitUtils.cmake
+++ b/Code/cmake/Modules/RDKitUtils.cmake
@@ -145,6 +145,7 @@ macro(rdkit_python_extension)
   CAR(RDKPY_NAME ${RDKPY_DEFAULT_ARGS})
   CDR(RDKPY_SOURCES ${RDKPY_DEFAULT_ARGS})
   if(RDK_BUILD_PYTHON_WRAPPERS)
+    find_package(Python3 COMPONENTS Development.Module)
     Python3_add_library(${RDKPY_NAME} MODULE ${RDKPY_SOURCES})
     set_target_properties(${RDKPY_NAME} PROPERTIES PREFIX "")
 
-- 
2.39.5 (Apple Git-154)
