diff --git a/Code/GraphMol/Substruct/SubstructDetails.h b/Code/GraphMol/Substruct/SubstructDetails.h
index 32d110771..58ecace97 100644
--- a/Code/GraphMol/Substruct/SubstructDetails.h
+++ b/Code/GraphMol/Substruct/SubstructDetails.h
@@ -23,7 +23,7 @@ class RecursiveStructureQuery;
 struct SubstructMatchParameters;

 namespace detail {
-struct RecursiveLocker {
+struct RDKIT_SUBSTRUCTMATCH_EXPORT RecursiveLocker {
   bool df_clearOnDestruct{true};
   std::vector<RecursiveStructureQuery *> locked;
   RecursiveLocker() = default;
