diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h b/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h
index b6811f23e..5c636a1c4 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h
@@ -383,7 +383,7 @@ RDKIT_SYNTHONSPACESEARCH_EXPORT void convertTextToDBFile(
  *
  * @return std::string
  */
-std::string formattedIntegerString(std::int64_t value);
+RDKIT_SYNTHONSPACESEARCH_EXPORT std::string formattedIntegerString(std::int64_t value);
 
 }  // namespace SynthonSpaceSearch
 }  // namespace RDKit
